src/components/feedback/snackbar/Snackbar.stories.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { useState } from 'react';
+
+import type { Meta, StoryObj } from '@storybook/react';
+import { expect, within, userEvent } from '@storybook/test';
+
+import { Snackbar, SnackbarProps } from './Snackbar';
+import { Button } from '../../basic/Button';
+
+const meta: Meta = {
+	title: 'Components/Feedback/Snackbar',
+	component: Snackbar,
+	parameters: {
+		layout: 'padded'
+	}
+} satisfies Meta<typeof Snackbar>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+const SnackbarStory = ({ open, label, ...rest }: SnackbarProps): React.JSX.Element => {
+	const [snack, setSnack] = useState(false);
+
+	return (
+		<>
+			<Button type="outlined" color="info" label="Click" onClick={() => setSnack(true)} />
+			<Snackbar onClose={() => setSnack(false)} label={label} open={open || snack} {...rest} />
+		</>
+	);
+};
+
+export const Success: Story = {
+	args: {
+		label: 'Lorem Ipsum dolor sit amet',
+		type: 'success',
+		open: false,
+		actionLabel: 'Ok',
+		disableAutoHide: false,
+		hideButton: false,
+		zIndex: 1000,
+		autoHideTimeout: 4000,
+		disablePortal: false,
+		singleLine: false
+	},
+	render: ({ open, label, ...rest }) => <SnackbarStory label={label} open={open} {...rest} />,
+	play: async ({ canvasElement }) => {
+		const canvas = within(canvasElement);
+		const { body } = canvasElement.ownerDocument;
+		const button = canvas.getByRole('button', {
+			name: /success/i
+		});
+		await expect(button).toBeVisible();
+		await userEvent.click(button);
+		expect(await within(body).findByText(/Lorem Ipsum dolor sit amet/i));
+	}
+};
+
+export const Info: Story = {
+	args: {
+		label: 'Lorem Ipsum dolor sit amet',
+		type: 'info',
+		actionLabel: 'Ok'
+	},
+	render: ({ label, open, ...rest }) => <SnackbarStory label={label} open={open} {...rest} />
```

Comment:
The descructure is not necessary (valid for every story)
```suggestion
	render: SnackbarStory
```

Reply:
I'm not sure, but I think I can't do this since `render` is a function which returns a component and it gives me an error.
Plus I haven't found any examples of it, they all do `render: () => component`

Reply:
SnackbarStory is a function that returns a component :smile: The fact that it gives you an error is that the meta object is typed wrongly. You have to remove the `: Meta` and leave only the satisfies.

The reason why I suggested it is that the anonymous component of the render is just a 1:1 pass-through of the props. You could remove label and open, and it would become
```tsx
render: ({ ...rest }) => <SnackbarStory {...rest} />
```

The following step would be removing the descructuring of the props
```tsx
render: (props) => <SnackbarStory {...props} />
```

Which could be simplified in
```
render: SnackbarStory
```

Deep diving a bit further, I've just realized that, since the rendered component is the same for every story, we should set the render field in the `meta` object, and then declare the story with just the args and possibily the other fields that differs between stories.